### PR TITLE
Asynchronous Expectations now work on Linux

### PR DESCRIPTION
**Pull request by ikesyo**

### CODE REVIEW

File: Sources/Nimble/Utils/Async.swift [hunk 1]
```diff
@@ -29,8 +33,13 @@ internal class AssertionWaitLock: WaitLock {
 
     func acquireWaitingLock(_ fnName: String, file: FileString, line: UInt) {
         let info = WaitingInfo(name: fnName, file: file, lineNumber: line)
+        #if _runtime(_ObjC)
+            let isMainThread = Thread.isMainThread
+        #else
+            let isMainThread = _CFIsMainThread()
```
Comment (ikesyo) [inline Sources/Nimble/Utils/Async.swift]: `Thread.isMainThread` is not implemented yet. 😢

File: Sources/Nimble/Utils/Async.swift [hunk 2]
```diff
@@ -1,10 +1,13 @@
+import CoreFoundation
+import Dispatch
 import Foundation
 
-#if _runtime(_ObjC)
-import Dispatch
+#if !_runtime(_ObjC)
+    private let NSEC_PER_SEC: CUnsignedLongLong = 1000000000
```
Comment (modocache) [inline Sources/Nimble/Utils/Async.swift]: This isn't defined on Linux? Maybe it's imported under a different name...? 😕

**Comment by modocache** [comment]
Wow, fantastic! Thanks, @ikesyo! I had one comment, but I'll merge this now and you can follow up later if you wish.

File: Sources/Nimble/Utils/Async.swift (hunk 2 above)
Comment (ikesyo) [inline Sources/Nimble/Utils/Async.swift]: I got the error without this 🤔 

```
/project/Sources/Nimble/Utils/Async.swift:315:93: error: use of unresolved identifier 'NSEC_PER_SEC'
            let interval = DispatchTimeInterval.nanoseconds(Int(pollInterval * TimeInterval(NSEC_PER_SEC)))
                                                                                            ^~~~~~~~~~~~
SwiftGlibc.CLOCKS_PER_SEC:1:12: note: did you mean 'CLOCKS_PER_SEC'?
public var CLOCKS_PER_SEC: Int { get }
           ^
```

Comment (ikesyo) [inline Sources/Nimble/Utils/Async.swift]: Okay, `import CDispatch` fixes the issue.

Comment (modocache) [inline Sources/Nimble/Utils/Async.swift]: If this is Swift 3, the new [namings](https://github.com/apple/swift-evolution/blob/master/proposals/0088-libdispatch-for-swift3.md#time) might make it look more like `DispatchTimeInterval.nanoseconds(1)`. I guess we'd need to move the `_runtime(_ObjC)` check lower to the callsite...?

Anyway, I'm not sure it's *better* to use `DispatchTimeInterval.nanoseconds(1)`, just thought I'd point out there's an alternative to `NSEC_PER_SEC` in Swift 3, so there might be a different way to do this here.

Comment (ikesyo) [inline Sources/Nimble/Utils/Async.swift]: I'd tried to replace some `TimeInterval` parameters to `DispatchTimeInterval`, but [this line](https://github.com/Quick/Nimble/blob/26f41da389a53c62b7f24ed55cd1e0e4bf665ad8/Sources/Nimble/DSL+Wait.swift#L31) made me unhappy at that time 😓

Comment (modocache) [inline Sources/Nimble/Utils/Async.swift]: Awesome! I didn't know about CDispatch, that's great!

Comment (ikesyo) [inline Sources/Nimble/Utils/Async.swift]: Submitted https://github.com/Quick/Nimble/pull/343. ✨

**Pull request merged**